cmd/xfn/spark/spark.go

```diff
@@ -154,51 +165,56 @@ func (c *Command) Run() error { //nolint:gocyclo // TODO(negz): Refactor some of
 	// recommended; 'run' is more for testing. In practice though run seems to
 	// work just fine for our use case.
 
-	//nolint:gosec // Executing with user-supplied input is intentional.
-	cmd := exec.CommandContext(ctx, c.Runtime, "--root="+root, "run", "--bundle="+b.Path(), runID)
-	cmd.Stdin = bytes.NewReader(req.GetInput())
+	var rsp *v1alpha1.RunFunctionResponse
+	if c.DryRun {
+		rsp = &v1alpha1.RunFunctionResponse{Output: req.GetInput()}
+	} else {
```

Comment:
Can you return early instead of having the else branch?

Reply:
Agreed - https://github.com/crossplane/crossplane/tree/master/contributing#return-early